### PR TITLE
[Vue] migrate reporting page directive

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #18429.**

Changes:
* Migrate piwik-reporting-page directive.
* Migrate reportingPageModel service to store.
* Moving interfaces around so different services can use them.

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
UI tests should be passing for this PR

File: plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    <ActivityIndicator
+      :loading="loading"
+    />
+    <div v-show="hasNoPage">{{ translate('CoreHome_NoSuchPage') }}</div>
+    <div
+      class="row"
+      v-for="widget in widgets"
+      :key="widget.uniqueId"
+    >
+      <Widget
```
Comment (sgiehl) [inline plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue:18]: I guess we need to add some special class or attribute here. We have some javascript looking for `[piwik-widget]`, which no longer does exist. See https://github.com/matomo-org/matomo/blob/b7bfb36a3d37e4d056ab3f0355672b85f318f3d9/plugins/CoreHome/javascripts/dataTable.js#L1934

This also causes the UI test failure where a tooltip is still visible.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Found one regression caused by this migration. Otherwise looks good to me

File: plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue (hunk 1 above)
Comment (diosmosis) [reply plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue:18]: :+1: nice catch, I'll put looking for selectors like this on my checklist.

**Comment by diosmosis** [comment]
@sgiehl updated

**Comment by diosmosis** [comment]
fixed another test failure + made some tweaks

**Review by peterhashair — COMMENTED** [review]
that looks good to me, just one test here failed, Travis didn't catch up.  
```
 Cannot read properties of undefined (reading 'toDisplayString')
  at plugins/CoreVue/polyfills/dist/MatomoPolyfills.min.js:7:135
```

**Comment by diosmosis** [comment]
@peterhashair I don't see this failure in https://app.travis-ci.com/github/matomo-org/matomo/builds/243453117 can you link to it?

**Comment by diosmosis** [comment]
@peterhashair nevermind, I see it now.

**Comment by sgiehl** [comment]
@diosmosis seems to work in general. But I got this JS error when a report page is loaded:
image

**Comment by diosmosis** [comment]
@sgiehl fixed that

**Review by sgiehl — APPROVED** [review]

**Pull request merged**